### ISSUE TITLE
use cgroup to get cpu core

### ISSUE THREAD

**Issue by hawkingrei** [pingcap/tidb#36894]
## Enhancement

use cgroup to get cpu core.

1. we can get the real CPU core, especially in docker.
2. we can dynamically get the real CPU core.

**Comment by tiancaiamao** [comment]
Try this one? https://pkg.go.dev/go.uber.org/automaxprocs

**Comment by hawkingrei** [comment]
> Try this one? https://pkg.go.dev/go.uber.org/automaxprocs

Yes, It is a good way but it is enabled when Application startup. But if TiDB in TiDB need VPA. It cannot change dynamically.
